pkg/gatewayserver/io/observability.go

```diff
@@ -46,12 +49,37 @@ var ioMetrics = &messageMetrics{
 		},
 		[]string{"protocol"},
 	),
+	droppedMessagesBufferFull: metrics.NewContextualCounterVec(
```

Comment:
I propose that we generalize this as just `droppedMessages` and add an `error` label.

The `[]string{"type"}` becomes `[]string{"type", "error"}` and then you put both labels (in order).

See https://github.com/TheThingsNetwork/lorawan-stack/blob/41a416cf99256c43195679f35b9d81f6adaf683b/pkg/applicationserver/io/web/observability.go#L72-L78